src/org/oidc/services/Webfinger.java

```diff
@@ -0,0 +1,158 @@
+package org.oidc.services;
+
+import com.auth0.msg.Message;
+import com.google.common.base.Strings;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.oidc.common.HttpMethod;
+import org.oidc.common.MissingRequiredAttribute;
+import org.oidc.common.ServiceName;
+import org.oidc.common.WebFingerError;
+import org.oidc.service.AbstractService;
+import org.oidc.service.base.HttpArguments;
+import org.oidc.service.data.State;
+import org.oidc.service.util.Constants;
+import org.oidc.service.util.URIUtil;
+
+public class Webfinger extends AbstractService {
+
+    Message response = new JRD();
+    ServiceName serviceName = ServiceName.WEB_FINGER;
+    private List<String> oidcIssuers;
+
+    /**
+     * Constants
+     */
+    private static final String ALLOW_HTTP_LINKS = "allowHttpLinks";
+
+
+    public Webfinger(ServiceContext serviceContext,
+                     State state,
+                     ServiceConfig config,
+                     List<String> oidcIssuers) {
```

Comment:
Yes, according to the WebFinger standard the link relation type (rel) can occur more then once in a request.
But we will never do that !
So there is no reason to make this a list.

Reply:
I don't now where oidcIssuer comes from but you should use linkRelationType or the shorter rel since that is what the WebFinger standard uses.